### PR TITLE
FINERACT-2421: Add unit tests for SQLBuilder

### PR DESCRIPTION
**Pull request by airajena**
# Description
Adds comprehensive unit tests for SQLBuilder - a security utility that builds SQL WHERE clauses without SQL injection risk.
## Changes
- 18 test methods covering:
  - Valid operators (=, <, >, <=, >=, <>, LIKE)
  - Validation (null, empty, invalid column names, invalid operators)
  - WhereLogicalOperator (AND, OR)
  - Edge cases (table prefixes, underscores, dashes)
## Related Issue
FINERACT-2421

### CODE REVIEW

**Review by Aman-Mittal — APPROVED** [review]
LGTM, It would be recommended to add JAVADOC for this

**Comment by airajena** [comment]
> LGTM, It would be recommended to add JAVADOC for this

Done

**Comment by Aman-Mittal** [comment]
> > LGTM, It would be recommended to add JAVADOC for this
> 
> Done

please also squash your commits into 1 commit

**Comment by airajena** [comment]
> > > LGTM, It would be recommended to add JAVADOC for this
> > 
> > 
> > Done
> 
> please also squash your commits into 1 commit

Done

**Comment by adamsaghy** [comment]
Please run:
`./gradlew spotlessApply spotbugsMain spotbugsTest checkstyleMain checkstyleTest`

**Comment by airajena** [comment]
> Please run: `./gradlew spotlessApply spotbugsMain spotbugsTest checkstyleMain checkstyleTest`

@adamsaghy  Running `./gradlew spotlessApply` locally results in a `SocketTimeoutException` when downloading the Eclipse Groovy formatter. 
The unit tests pass locally. Could you please check if the CI failures are related to this network issue and potentially re-trigger the workflows?

**Comment by adamsaghy** [comment]
> > Please run: `./gradlew spotlessApply spotbugsMain spotbugsTest checkstyleMain checkstyleTest`
> 
> @adamsaghy Running `./gradlew spotlessApply` locally results in a `SocketTimeoutException` when downloading the Eclipse Groovy formatter. The unit tests pass locally. Could you please check if the CI failures are related to this network issue and potentially re-trigger the workflows?

Its not timeout issue at CI. Please make sure to run locally this command till it got successfully executed!

**Comment by adamsaghy** [comment]
```
For more on this, please refer to https://docs.gradle.org/8.14.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.
168 actionable tasks: 147 executed, 5 from cache, 16 up-to-date
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':fineract-core:spotlessJavaCheck'.
> The following files had format violations:
      src/test/java/org/apache/fineract/infrastructure/security/utils/SQLBuilderTest.java
          @@ -25,8 +25,8 @@
           /**
           ·*·Unit·tests·for·{@link·SQLBuilder}.
           ·*·<p>
          -·*·Tests·the·SQL·WHERE·clause·builder·utility·that·prevents·SQL·injection
          -·*·by·validating·column·names,·operators,·and·using·parameterized·queries.
          +·*·Tests·the·SQL·WHERE·clause·builder·utility·that·prevents·SQL·injection·by·validating·column·names,·operators,·and
          +·*·using·parameterized·queries.
           ·*·</p>
           ·*/
           class·SQLBuilderTest·{
  Run './gradlew spotlessApply spotbugsMain spotbugsTest checkstyleMain checkstyleTest' to fix these violations.

* Try:
> Run with --stacktrace option to get the stack trace.
```

**Comment by adamsaghy** [comment]
Please fix it:
https://github.com/apache/fineract/pull/5277#issuecomment-3727775893

**Comment by airajena** [comment]
> Please fix it: [#5277 (comment)](https://github.com/apache/fineract/pull/5277#issuecomment-3727775893)

Fixed
